src/formatDate.js

```diff
@@ -8,7 +8,38 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
-}
+  const partsDate = date.split(fromFormat[3]);
+  let year = 0;
+
+  if (fromFormat.includes('YY')) {
+    year = parseInt(partsDate[fromFormat.indexOf('YY')]);
+    year = year < 30 ? 2000 + year : 1900 + year;
+  } else {
+    year = partsDate[fromFormat.indexOf('YYYY')];
+  }
+
+  const month = partsDate[fromFormat.indexOf('MM')];
+  const day = partsDate[fromFormat.indexOf('DD')];
+
+  const newSper = toFormat[3];
```

Comment:
Similar to the issue with `fromFormat`, the code assumes that the separator is at index 3 in the `toFormat` array. This might not be correct for all formats. Consider a more flexible approach to determine the separator.